tests/TestCase.php

```diff
@@ -81,4 +83,21 @@ function (
             );
         }
     }
+
+    protected static function getFixerByName(string $name): AbstractFixer
```

Comment:
```suggestion
    protected static function getFixerByName(string $name): FixerInterface
```

Reply:
Nah, I like CI not failing 😉

Reply:
If you already know that CI would be failing, consider to share the reason. Otherwise you either provoke to original requester to ask "why?" or to spend time to get the knowledge you already have.

The only place I see failing is SCA for running `->getConfigurationDefinition()` in `assertPHPUnitVersionIsLargestAllowed`. That single usage is IMHO not enough to expose everything (`AbstractFixer`) to everywhere (exposed in `TestCase`). I would strongly advise to expose only minimum need (`FixerInterface`) only when needed (eg via trait).

And to be honest that fail is fair - right now, test assumes that given rule is configurable, which may not be the case.

Reply:
> which may not be the case

but it is... I'm very confused, did the YAGNI become some good practice recently, and I have not got a memo?

In the thread above you want me to add a check for something that never happens, here you agree that an assumption (which is the assumption only if the return type is changed to `FixerInterface`, with `AbstractFixer` it is not) that stands for over 3.

Reply:
This one is no longer an issue after https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7961 - we should gowith that change instead of this discussion.